### PR TITLE
Fix lint errors in jetpack-connect install step

### PR DESCRIPTION
**Pull request by sirreal**
Update to es6 class.
Fix usage of `this.translate`.
Add `propTypes`

Lint errors must be fixed before jetpack-connect can be moved (#12991).

To test:

* Use this branch.
* You must have a disconnected site with Jetpack not installed.
* Go to /jetpack/connect
* Enter your site's URL.
* Follow the steps, returning to /jetpack/connect after each step (install, activate, connect).
* Ensure there are no regressions. Watch the console (preserve log) for any errors/warnings.
* Bonus points for testing with old example (must use Jetpack `<4.2`) which uses a different connect example component.

### CODE REVIEW

File: client/signup/jetpack-connect/install-step.jsx [hunk 1]
```diff
@@ -15,83 +16,100 @@ import JetpackExampleConnect from './example-components/jetpack-connect';
 
 const NEW_INSTRUCTIONS_JETPACK_VERSION = '4.2.0';
 
-export default React.createClass( {
-	displayName: 'JetpackInstallStep',
+class JetpackInstallStep extends Component {
```
Comment (tyxla) [inline client/signup/jetpack-connect/install-step.jsx]: Usually our linter will complain about empty lines in the beginning of code blocks. For some reason that does not happen in classes. But let's remove that one regardless.

File: client/signup/jetpack-connect/install-step.jsx [hunk 2]
```diff
@@ -15,83 +16,100 @@ import JetpackExampleConnect from './example-components/jetpack-connect';
...
 
-	confirmJetpackInstalled( event ) {
+	static propTypes = {
+		currentUrl: PropTypes.string.isRequired,
+		confirmJetpackInstallStatus: PropTypes.func.isRequired,
+		jetpackVersion: PropTypes.string,
+		onClick: PropTypes.func.isRequired,
```
Comment (tyxla) [inline client/signup/jetpack-connect/install-step.jsx]: IMHO `onClick` should be optional; having it as required feels a little odd. Instead, I'd expect to see it specified as a non-required prop, but also to have a `onClick: noop` default prop value defined.

File: client/signup/jetpack-connect/install-step.jsx [hunk 3]
```diff
@@ -15,83 +16,100 @@ import JetpackExampleConnect from './example-components/jetpack-connect';
 
 const NEW_INSTRUCTIONS_JETPACK_VERSION = '4.2.0';
 
-export default React.createClass( {
-	displayName: 'JetpackInstallStep',
+class JetpackInstallStep extends Component {
 
-	confirmJetpackInstalled( event ) {
+	static propTypes = {
+		currentUrl: PropTypes.string.isRequired,
```
Comment (tyxla) [inline client/signup/jetpack-connect/install-step.jsx]: In that case I think we can freely make this prop non-required and set its default prop value as an empty string. What do you think?

File: client/signup/jetpack-connect/install-step.jsx (hunk 1 above)
Comment (sirreal) [inline client/signup/jetpack-connect/install-step.jsx]: 👍  a2b9581e2

Eventually I'll internalize this and you won't have to correct me on every PR 🙂

File: client/signup/jetpack-connect/install-step.jsx (hunk 3 above)
Comment (sirreal) [inline client/signup/jetpack-connect/install-step.jsx]: 👍  7f5a9392b

Sounds good, this is actually what some components already do down the line.

File: client/signup/jetpack-connect/install-step.jsx (hunk 2 above)
Comment (sirreal) [inline client/signup/jetpack-connect/install-step.jsx]: 👍  7f5a9392b

**Comment by sirreal** [comment]
I'll need to come back to this. `propType` may be incorrect for `JetpackVersion`, seeing this warning when latest Jetpack (`4.9`) is installed but not activated.

```
Warning: Failed prop type: Invalid prop `jetpackVersion` of type `boolean` supplied to `JetpackInstallStep`, expected `string`.
```

**Comment by tyxla** [comment]
> I'll need to come back to this. propType may be incorrect for JetpackVersion, seeing this warning when latest Jetpack (4.9) is installed but not activated.

Looking at the endpoint, it appears that the version can be either `string` or `boolean`, so `PropTypes.oneOfType( [ PropTypes.object, PropTypes.string ] )` should be fine to fix that.

**Comment by sirreal** [comment]
@tyxla I've gotten `false` several times, but I haven't been able to consistently reproduce it. I'm curious about what the conditions are to get `false` for the jetpackVersion prop.

Added `bool` to valid propType in 80f0b8e09.

This should be ready to ship now.

**Pull request merged**